test/wasi/c/sock.c

```diff
@@ -0,0 +1,17 @@
+#include <sys/socket.h>
+#include <stddef.h>
+#include <errno.h>
+#include <assert.h>
+#include <stdio.h>
+
+int main(void) {
+
+  int fd = 0 ;
+  socklen_t addrlen = 0;
+  int flags = 0;
+  int ret = accept(0, NULL, &addrlen);
+  assert(ret == -1);
+  assert(errno == 58);
```

Comment:
You should be able to compare against error codes here. [[example](https://github.com/nodejs/node/blob/16fbfbe152ff14247e0c6be9b38a8dcea5b6cef0/test/wasi/c/notdir.c#L8)]